### PR TITLE
Autopilot telemetry

### PR DESCRIPTION
**Pull request by makhov**
Signed-off-by: Alexey Makhov <amakhov@mirantis.com>

## Description

Send events for the autopilot update process

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

## Checklist:

- [ ] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [ ] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pkg/autopilot/common/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	OldStatus string
+	NewStatus string
+}
+
+func ReportEvent(evt *Event) error {
+	if reporter == nil {
+		return nil
+	}
+	return reporter.Enqueue(analytics.Track{
+		Event:      eventName,
```
Comment (soider) [inline pkg/autopilot/common/telemetry.go:36]: let's add the cluster id thing, to trace somehow the events

File: pkg/autopilot/common/telemetry.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		Properties: evt.asProperties(),
+	})
+}
+
+func (e *Event) asProperties() analytics.Properties {
+	return analytics.Properties{
+		"oldStatus": e.OldStatus,
+		"newStatus": e.NewStatus,
```
Comment (soider) [inline pkg/autopilot/common/telemetry.go:45]: cluster id field might be added here as well

Comment (makhov) [reply pkg/autopilot/common/telemetry.go:46]: Great catch, thank you. Fixed

File: pkg/autopilot/common/telemetry.go (hunk 1 above)
Comment (makhov) [reply pkg/autopilot/common/telemetry.go:37]: Done

**Review by soider — APPROVED** [review]

**Pull request merged**